core/lib/basic_types/src/lib.rs

```diff
@@ -78,32 +78,15 @@ impl TryFrom<U256> for AccountTreeId {
 
 /// ChainId in the ZkSync network.
 #[derive(Copy, Clone, Debug, Serialize, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct L2ChainId(pub u64);
+pub struct L2ChainId(u64);
 
 impl<'de> Deserialize<'de> for L2ChainId {
     fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
     where
         D: Deserializer<'de>,
     {
         let s: String = Deserialize::deserialize(deserializer)?;
-
-        // Parse the string as a U256
-        // try to parse as decimal first
-        let u64 = match U64::from_dec_str(&s) {
-            Ok(u) => u,
-            Err(_) => {
-                // try to parse as hex
-                s.parse::<U64>().map_err(de::Error::custom)?
-            }
-        };
-
-        if u64.as_u64() > L2ChainId::max().0 {
-            return Err(de::Error::custom(format!(
-                "Too big chain ID. MAX: {}",
-                L2ChainId::max().0
-            )));
-        }
-        Ok(L2ChainId(u64.as_u64()))
+        s.parse().map_err(de::Error::custom)
```

Comment:
why are you not checking the max value anymore?

Reply:
MAX is checked in FromStr, i.e parse() is invoked

Reply:
Let's have a few unittests.